python/lsst/daf/butler/cli/cmd/commands.py

```diff
@@ -159,12 +161,61 @@ def query_collections(*args, **kwargs):
 @options_file_option()
 def query_dataset_types(*args, **kwargs):
     """Get the dataset types in a repository."""
-    print(yaml.dump(cli_handle_exception(queryDatasetTypes, *args, **kwargs), sort_keys=False))
+    print(yaml.dump(cli_handle_exception(script.queryDatasetTypes, *args, **kwargs), sort_keys=False))
 
 
 @click.command()
 @repo_argument(required=True)
 @click.argument('dataset-type-name', nargs=1)
 def remove_dataset_type(*args, **kwargs):
-    """Remove a dataset type definition from a reopsitory."""
-    cli_handle_exception(removeDatasetType, *args, **kwargs)
+    """Remove a dataset type definition from a repository."""
+    cli_handle_exception(script.removeDatasetType, *args, **kwargs)
+
+
+@click.command()
+@repo_argument(required=True)
+@glob_argument(help="GLOB is one or more glob-style expressions that fully or partially identify the "
+                    "dataset types to be queried.")
+@click.option("--collections",
+              help=unwrap("""One or more expressions that fully or partially identify the collections to
+                          search for datasets.If not provided all datasets are returned."""),
+              multiple=True,
+              metavar=typeStrAcceptsMultiple,
+              callback=split_commas)
+@click.option("--where",
+              help=unwrap("""A string expression similar to a SQL WHERE clause. May involve any column of a
+                          dimension table or a dimension name as a shortcut for the primary key column of a
+                          dimension table."""))
+@click.option("--deduplicate",
```

Comment:
I don't think deduplicate should be a user interface option at all. Why would someone ever want to see a duplicate row. All it does is add confusion to the output.  I think that you should use the deduplicate flag internally where you can and if you can't you need to throw the per-dataset type refs into a set and then extract them from the set sorted by ref.id.

Reply:
Maybe this needs a new name, but it is not at all equivalent to deduplicating results after the fact.  `deduplicate=True` says to only return one result for each datasetType+dataId combination across all searched collections, searching them in order.

Reply:
Even more important not to confuse the output here then where we get identical answers in the output and no-one knows why. Are you saying that I got two rows because a dataset happened to be in two collections but since the collection is not known to the DatasetRef we can't tell why we had two rows?

Reply:
I think it's somewhere in between, because DatasetRef does know the dataset's RUN-type collection, so both that and the dataset_id will always be printed, and the fact that they're actually different and where they're coming from will be apparent.  That should be the common case.

But it is also be possible to get the exact same dataset multiple times because both the RUN it's in and a TAGGED collection it's in are being queried.  Providing that information about why it appeared twice is hard right now, but just removing that duplication is as simple as stuffing the query results in a `set`, and that seems like the right thing to do anyway.